src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -78,7 +56,7 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
 
     $ids = [];
 
-    if ($this->configuration['handler_settings']['field_mode'] == 'admin') {
+    if ($this->isGroupAdmin()) {
```

Comment:
> Can we remove the field_mode stuff?

No, this part is calculated on the fly and helps us create each widget inside our complex widget (e.g. the "my groups"  and "other groups). Probably worth having a hangout about it, and I'll guide you through the D7 code to understand the logic.